docs/debug.md

```diff
@@ -0,0 +1,650 @@
+
+```
+$ snappy-debug.strace cat /etc/shadow
+...
+open("/etc/shadow", O_RDONLY)           = -1 EACCES (Permission denied)
+...
+```
+
+This indicates that the open system call failed with the given error number.
+
+### ltrace
+
+The ltrace tool is similar to strace. The difference is that it tracks
+library calls instead of system calls.
+
+The command output looks like this:
+
+```
+$ snappy-debug.ltrace /etc/shadow
+__libc_start_main(0x401a40, 2, 0x7ffc27e9f058, 0x408b40 <unfinished ...>
+getpagesize()                                    = 4096
+strrchr("cat", '/')                              = nil
+setlocale(LC_ALL, "")                            = "en_US.UTF-8"
+bindtextdomain("coreutils", "/usr/share/locale") = "/usr/share/locale"
+textdomain("coreutils")                          = "coreutils"
+__cxa_atexit(0x402ac0, 0, 0, 0x736c6974756572)   = 0
+getopt_long(2, 0x7ffc27e9f058, "benstuvAET", 0x409280, nil) = -1
+__fxstat(1, 1, 0x7ffc27e9eea0)                   = 0
+open("/etc/shadow", 0, 0400000)                  = -1
+__errno_location()                               = 0x7f1ed2527690
+```
+
+## Important system logs
+
+Snappy Ubuntu system logs are mapped to the classic location of those logs.
+The most important log file to look at is:
+
+    /var/log/syslog
+
+The syslog is particularly useful since kernel logs, launcher output, service
+status, system logs and confinement violations (these are covered in depth
+elsewhere) all get logged there.
+
+Further, systemd has log dump facility to look at all system and snap
+services. The "snappy service" command provides an easy way to monitor and
+inspect snap services. To get the status of all snap services run "snappy
+service status". To inspect the logs run "snappy service logs". Both commands
+can take a snap name to limit to a specific snap and also `<snapname>` to
+limit to only a specific service of a snap (ie for the shout service in
+`shout.sergiusens` run: `sudo snappy service logs shout`).
+
+
+## Debugging binaries
+
+### Installing binaries
+In order to test a new snap on a Snappy system you need to install it first.
+This is called sideloading and it can be done via
+`snappy-remote --url=ssh://ubuntu@webdm.local/ install snapname.snap`.
+Alternatively the snap can be copied via scp into the Snappy system and
+installed via `sudo snappy install snapname.snap`.
+
+Note that you have to use the `--allow-unauthenticated` tag for installing
+unsigned snaps: `sudo snappy install --allow-unauthenticated snapname.snap`.
+
+### Binary names on CLI
+
+The name of the binaries will be prefixed with the snap package name. This
+ensures that there are never any namespace collision in the binary names. For
+example, the snap `pastebinit.mvo` contains a `pastebinit` script. The binary
+on disk will be called `pastebinit.pastebinit`. The exception of this rule are
+frameworks which may have short binary names. E.g. the docker package contains
+a docker binary.
```

Comment:
Perhaps the framework sentence should be removed considering that frameworks as we know them will not exist in 16.04.